fluss-server/src/test/java/org/apache/fluss/server/RpcServiceBaseTest.java

```diff
@@ -0,0 +1,758 @@
+
+        // Execute test
+        CompletableFuture<PartitionMetadata> future =
+                RpcServiceBase.getPartitionMetadataFromZkAsync(testPartitionPath, mockZkClient);
+
+        // The entry should be added to the map
+        assertThat(getPendingPartitionMetadataFutureMap()).hasSize(1);
+
+        // Wait for completion
+        future.get();
+
+        // After completion, the entry should be removed from the map
+        assertThat(getPendingPartitionMetadataFutureMap()).isEmpty();
+    }
+
+    @Test
+    void testGetPartitionMetadataFromZkAsync_MapEntryRemovedOnFailure() throws Exception {
+        // Mock ZooKeeper to throw exception
+        when(mockZkClient.getPartition(testTablePath, "partition1"))
+                .thenThrow(new RuntimeException("ZK connection failed"));
+
+        // Check that the map is initially empty
+        assertThat(getPendingPartitionMetadataFutureMap()).isEmpty();
+
+        // Execute test
+        CompletableFuture<PartitionMetadata> future =
+                RpcServiceBase.getPartitionMetadataFromZkAsync(testPartitionPath, mockZkClient);
+
+        // The entry should be added to the map
+        assertThat(getPendingPartitionMetadataFutureMap()).hasSize(1);
+
+        // Wait for completion and expect exception
+        assertThatThrownBy(future::get)
+                .isInstanceOf(ExecutionException.class)
+                .hasCauseInstanceOf(FlussRuntimeException.class);
+
+        // After failure, the entry should be removed from the map
+        assertThat(getPendingPartitionMetadataFutureMap()).isEmpty();
+    }
+
+    @Test
+    void testProcessMetadataRequest_TableFromCache() throws Exception {
+        // Setup mocks
+        MetadataRequest request = new MetadataRequest();
+        PbTablePath pbTablePath = new PbTablePath().setDatabaseName("db").setTableName("table");
+        request.addTablePath().copyFrom(pbTablePath);
+
+        String listenerName = "default";
+        Session session = createTestSession();
+        Authorizer authorizer = mock(Authorizer.class);
+        ServerMetadataCache metadataCache = mock(ServerMetadataCache.class);
+        MetadataFunctionProvider functionProvider = mock(MetadataFunctionProvider.class);
+
+        when(authorizer.isAuthorized(any(), any(), any())).thenReturn(true);
+        TablePath tablePath = TablePath.of("db", "table");
+        TableInfo partitionTableInfo =
+                TableInfo.of(
+                        tablePath,
+                        testTableId,
+                        1,
+                        DATA1_PARTITIONED_TABLE_DESCRIPTOR,
+                        System.currentTimeMillis(),
+                        System.currentTimeMillis());
+        TableMetadata tableMetadata =
+                new TableMetadata(partitionTableInfo, Collections.emptyList());
+        when(functionProvider.getTableMetadataFromCache(eq(tablePath)))
+                .thenReturn(Optional.of(tableMetadata));
```

Comment:
Please do no use Mockito, it increases maintenance overhead in the long term, which should be avoided. In this case, we can simply pass ` authorizer` as `null`, and create a testing implementation for `MetadataFunctionProvider` and `ServerMetadataCache`.